content/js/handling-errors.md

```diff
@@ -2,25 +2,47 @@ All asynchronous methods in `skygear` return [Promise](https://www.promisejs.org
 
 ## Basic flow
 
-```js
-skygear.signup('johndoe', 'john.doe@example.com', 'verysecurepassword').then(() => {
+``` javascript
+skygear.signupWithUsername('johndoe', 'verysecurepassword').then(() => {
   // user logged in successfully
 }, (error) => {
-  if (error.code == skygear.Error.DUPLICATED) {
-    console.log('username has been taken');
-  } else {
-    console.log('unexpected error', error.code)
+  if (error.code === 109) {
+    alert('username or email already exists');
   }
+  console.error(error);
 });
 
 ```
 
 ## Ignoring errors
 
-```js
+``` javascript
 // omit the error callback
-skygear.signup('johndoe', 'john.doe@example.com', 'verysecurepassword').then(() => {
+skygear.signupWithUsername('johndoe', 'verysecurepassword').then(() => {
```

Comment:
I am not sure about the intention here for "ignoring errors". I don't think we should ignore a rejection from a promise